File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file230.f90
subroutine kernel_mvt(n, x1, x2, y1, y2, a)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n) :: x1
DATA_TYPE, dimension(n) :: y1
DATA_TYPE, dimension(n) :: x2
DATA_TYPE, dimension(n) :: y2
integer :: n
integer :: i, j

!$pragma scop
do i = 1, _PB_N
do j = 1, _PB_N
x1(i) = x1(i) + (a(j, i) * y1(j))
end do
end do
do i = 1, _PB_N
do j = 1, _PB_N
x2(i) = x2(i) + (a(i, j) * y2(j))
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file230.cpp
void kernel_mvt(int n, double* x1, double* x2, double* y1, double* y2, double** a)
{
 int i, j;
 for (i = 0; i < n; i++)
 for (j = 0; j < n; j++)
 x1[i] += a[j][i] * y1[j];
 for (i = 0; i < n; i++)
 for (j = 0; j < n; j++)
 x2[i] += a[i][j] * y2[j];
}
